### PR TITLE
Redo

### PR DESCRIPTION
**Pull request by murrman95**
pls

### CODE REVIEW

**Comment by yongheng1991** [comment]
Hi,
Thanks very much for the implementation of the EM routing. I used to have the same idea but never got time to try. Unfortunately, I could not test it now cause I am on vacation until the end of Jan 2020. I would appreciate it if you could share with me the performance of the EM routing compared with the original one. 

Also, thanks for the modification of importing the updated Chamfer distance package.
Best

**Comment by murrman95** [comment]
Hey Yongheng,

First off, great job with the repo and paper, but we had problems getting it to run on our machines so we resorted to running on google collab. Perhaps more information about how you got it to run on your machine would be helpful? From the information given, it should have worked on our machines.

Honestly. I didn't mean to try to try to merge with you. This was part of my (and @GuillaumeDufau)'s Computer Vision project, so I only meant to merge with my original fork. But anyway, our solution is a little hacky. There were existing EM Routing solutions for MNIST-type datasets, so they were used implemented using 2-dimensional convolutions. To get the point-net representation to work, just re imagined the 1D convolution as 2D convolution with kernel size 1 and an image size of 1x1. 

But anyway, we didn't do any sort of lambda-scheduler for the EM, so there was no hyperparameter tuning for that. From the settings included in our implementation, we got pretty much the same loss curve during training, the EM routing was actually much faster for us (45 mins for 3 epochs on a google collab instance instead of 55 minutes). Also, our solution constricts the LatentCaps layer to contain 32 capsules with size 32 vectors, so perhaps our implementation isn't so comparable. There is still a bug to find to be able to use the pre-defined num_latent_caps and latent_vector_size (the output dimensions from the Primary Caps isn't correct).

If you're interested in merging our results, there is a newer commit on our branch which has the final scripts. We could collapse our commits to make the git log cleaner.
